track_health.rb

```diff
@@ -1,7 +1,40 @@
 require_relative 'encounter'
+require 'json'
 
-encounter_file_location = 'data/encounters/encounter-1.json'
-encounter = Encounter.new(encounter_file_location)
+ENCOUNTERS_DIRECTORY = 'data/encounters'.freeze
 
-puts "Encounter Name: #{encounter.name}"
-puts "Monster Names: #{encounter.monster_names.join(', ')}"
+def put_encounter_name(encounter_path)
+  encounter_being_checked = JSON.parse(File.read("#{encounter_path}"))
+  encounter_being_checked["name"]
+end
+
+def get_integer_in_range(maximum, minimum:0)
+  input = gets.strip
+  if input.to_i.to_s != input
+    puts "Please enter a whole number"
+    get_integer_in_range(maximum, minimum:minimum)
+  elsif (input.to_i < minimum) or (input.to_i > maximum)
+    puts "please enter a number between #{minimum} and #{maximum} (inclusive)."
+    get_integer_in_range(maximum, minimum:minimum)
+  else
+    input.to_i
+  end
+end
+
+filenames = Dir["#{ENCOUNTERS_DIRECTORY}/*.json"]
+
+encounter_names = filenames.map {
+  |file_path|
+  put_encounter_name(file_path)
+}
+
+puts "Please select an encounter from the list:"
+
+encounter_names.length.times {
+  |item_index|
+  puts "#{item_index}: #{encounter_names[item_index]}"
+}
```

Comment:
This should use `do`/`end` block syntax.

Use `.each.with_index` so that you get access to the item you're iterating over.

```ruby
encounter_names.each.with_index do |encounter_name, index|
   puts "#{index}: #{encounter_name}"
end
```

Reply:
Let's change this to start at `1`.